Sources/macro.js

```diff
@@ -257,7 +257,7 @@ export function algo(publicAPI, model, numberOfInputs, numberOfOutputs) {
   }
 
   function getInputData(port = 0) {
-    if (!model.inputData[port]) {
+    if (!model.inputData[port] && model.inputConnection[port]) {
```

Comment:
It seemed like `getInputData` was trying to use cached data. You're saying it should always invoke the connection if one exists?